olog/util.py

```diff
@@ -30,5 +30,16 @@ def ensure_time(time):
     return time.isoformat(sep=' ', timespec='milliseconds')
 
 
+def check_name(name):
+    if not isinstance(name, str):
+        raise TypeError(f'name should be a str, {name} is not a str.')
+    return name
+
+
+def simplify_attr(d):
```

Comment:
Except in special circumstances, functions should not modify their inputs. Could you create a shallow copy of `d`, modify that, and return it?

Reply:
Try to understand more details.
>functions should not modify their inputs.

Why? Is it anti-pattern? My thinking is "if function only update small piece of input, it should try to modify original input instead of copy and modify new one". Also code is less without copy.

Reply:
It’s a fair question which I am glad you asked. Yes, I would call it an anti-pattern. The reason that mutating inputs is dangerous is that it can lead to unexpected behavior. Perhaps the same mutable object is passed to two functions. If one mutates it in place, the other may not receive what it expects.

You are completely right that mutating in place is more efficient, but performance concerns have to be considered in context. A network request to the Olog will always be much more costly than a shallow copy of a small dict. Therefore, avoiding mutating inputs—a rule of thumb that avoids confusing bugs that cost developer and user time—is worthwhile in this case.